### PR TITLE
feat(indexer): Implement presets for toml file

### PR DESCRIPTION
**Pull request by roninjin10**
- Implement presets for toml file for known chains like base base goerli op op goerli zora and zora goerli
- Add full test suite since this code is ready to go outside of possibly adding more contracts

### CODE REVIEW

**Comment by roninjin10** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #6775** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6775" target="_blank">Graphite</a>  👈
    * **PR #6769** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6769" target="_blank">Graphite</a> 
      * **PR #6772** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6772" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6775?utm_source=stack-comment).

File: indexer/config/presets.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		L1ERC721Bridge:         common.HexToAddress("0x8DD330DdE8D9898d43b4dc840Da27A07dF91b3c9"),
+	},
+	// Base Mainnet
+	8453: {
+		OptimismPortal:         common.HexToAddress("0x49048044D57e1C92A77f79988d21Fa8fAF74E97e"),
+		L2OutputOracle:         common.HexToAddress("0x56315b90c40730925ec5485cf004d835058518A0"),
+		L1CrossDomainMessenger: common.HexToAddress("0x866E82a600A1414e583f7F13623F1aC5d58b0Afa"),
+		L1StandardBridge:       common.HexToAddress("0x3154Cf16ccdb4C6d922629664174b904d80F2C35"),
+		// FIXME update this to the correct address
+		L1ERC721Bridge: common.HexToAddress("0x0000000000000000000000000000000000000000"),
```
Comment (roninjin10) [inline indexer/config/presets.go:34]: Gonna have to find these via this other_bridge property. Since we aren't using anything other than OP atm I skipped for now 
image

Comment (roninjin10) [reply indexer/config/presets.go:34]: Could consider not specifying it and just index it but seems pointless to add that complexity when onchain/onrpc config is coming

File: indexer/config/config.go [hunk 2]
```diff
@@ -85,19 +89,25 @@ type MetricsConfig struct {
...
 
-	// Replace environment variables.
 	data = []byte(os.ExpandEnv(string(data)))
 
-	// Decode the TOML data.
 	if _, err := toml.Decode(string(data), &conf); err != nil {
 		return conf, err
 	}
 
+	if conf.Chain.Preset != 0 {
```
Comment (roninjin10) [inline indexer/config/config.go:113]: The toml library seems to set everything to 0 if it isn't provided. I looked in the docs for a way to make it explicitly nil but couldn't (quickly) find one. This passes all the tests this implementation though.

Would be nice if we could throw an error when required stuff is not provided fully off of the type. This will throw an error if a unknown config option is passed in though. Maybe worth noting in a github issue that the toml is defaulting a lot of stuff we would prefer to not default

Comment (roninjin10) [reply indexer/config/config.go:113]: Maybe we should make a ticket to provide extra validation to fail early here? We could 
- check all the contract addresses exist on the chain on startup
- - Check that the rpc is up
- - Check that every field is uo

File: indexer/config/presets.go (hunk 1 above)
Comment (hamdiallam) [reply indexer/config/presets.go:34]: I'm also open to simply removing this contract address from the list until we get around to supporting the 721 bridge

File: indexer/config/config.go (hunk 2 above)
Comment (hamdiallam) [reply indexer/config/config.go:113]: what happens if you make the field a pointer to an int? Will it initialize it to `nil` if it's not present?

File: indexer/indexer.toml [hunk 3]
```diff
@@ -20,3 +22,11 @@ port = 8080
 host = "127.0.0.1"
 port = 7300
 
+[logger]
```
Comment (hamdiallam) [inline indexer/indexer.toml:25]: I think we're going with removing this from config? ffti if you're going to make undo change in a different PR

**Review by hamdiallam — APPROVED** [review]
LGTM! Left a comment about how you might get around toml deserializattion for values not present

File: indexer/indexer.toml (hunk 3 above)
Comment (roninjin10) [reply indexer/indexer.toml:25]: Yea will remove

File: indexer/config/config.go (hunk 2 above)
Comment (roninjin10) [reply indexer/config/config.go:113]: will give it a try
